label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRuleData data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  MatchingCriteria matching_criteria = 2;
+  ApplicationCriteria application_criteria = 3;
+
+  message MatchingCriteria {
+    Condition condition = 1;
+
+    message Condition {
+      oneof condition {
+        LeafCondition leaf_condition = 1;
+        CompositeCondition composite_condition = 2;
+      }
+    }
+
+    message CompositeCondition {
+      LogicalOperator operator = 1;
+      repeated Condition children = 2;
+
+      enum LogicalOperator {
+        LOGICAL_OPERATOR_UNSPECIFIED = 0;
+        LOGICAL_OPERATOR_AND = 1;
+        LOGICAL_OPERATOR_OR = 2;
+      }
+    }
+
+    message LeafCondition {
+      StringCondition key_condition = 1;
+      oneof condition {
+        StringCondition string_condition = 2;
+        UnaryCondition unary_condition = 3;
+        JsonCondition json_condition = 4;
+      }
+    }
+
+    message StringCondition {
+      Operator operator = 1;
+      string value = 2;
+
+      enum Operator {
+        OPERATOR_UNSPECIFIED = 0;
+        OPERATOR_EQUALS = 1; // operator to check if the key exists, and value is equal to the provided value
+        OPERATOR_MATCHES_REGEX = 2; // operator to check if the key exists, and value matches the provided regex value
+      }
+    }
+
+    message UnaryCondition {
+      Operator operator = 1;
+
+      enum Operator {
+        OPERATOR_UNSPECIFIED = 0;
+        OPERATOR_EXISTS = 1; // checks for the existence for the key
+      }
+    }
+
+    message JsonCondition {
+      string json_path = 1; // path to json structure
+      oneof value_condition {
+        StringCondition string_condition = 2;
+        UnaryCondition unary_condition = 3;
+      }
+    }
+  }
+
+  message ApplicationCriteria {
+    string entity_type = 1; // captures the entity type - "API", "Service"
+    Action action = 2;
+
+    message Action {
+      Operation operation = 1;
+      oneof value {
+        string static_label_id = 2;
+        DynamicLabel dynamic_label = 3;
```

Comment:
so dynamic labels are not defined in the system with an id ? what is the difference between static_label_id and dynamic_label ?

Reply:
All the labels will be associated with an entity using an id. That is same for static or dynamic labels.

Please note that static labels pre-exists in the system and will have id already associated with it. Dynamic labels are created at runtime. If it doesn't exist, then it is created along with its id. Being dynamic in nature, the api definition only captures how to generate them.